### DISCUSSION TITLE
Allow local paths to be passed to ui.download, ui.image, etc

### DISCUSSION THREAD

**Discussion by rodja** [zauberzeug/nicegui#833] [Ideas / Feature Requests]
It would be nice if users could directly provide a file path in elements. Currently it's only allowed to pass urls. So files need to be made available before via [ui.add_static_files](https://nicegui.io/documentation#add_static_files_) or similar.

If we would detect file paths automatically we should not use ui.add_static_files internally because it will allow access to the whole folder and thereby poses a security risk. It may be better to create a custom route with FastAPI delivering the data with [FileResponse](https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse).

**Comment by falkoschindler** [comment]
Instead of automatically distinguishing URL paths and file paths (maybe based on datatype `str` vs. `Path`?), we could simply use different arguments like `ui.download(url: Optional[str] = None, *, local_path: Optional[str] = None, ...)`. And we could also add `content: Optional[Union[bytes, str]]` to accept the content directly. (Or should it be `bytearray` instead of `bytes`?)
